src/config.rs

```diff
@@ -609,25 +614,28 @@ impl IntoIterator for Address {
 
     fn into_iter(self) -> Self::IntoIter {
         match self {
-            // TODO(https://github.com/istio/ztunnel/issues/1131) allow only v4
-            Address::Localhost(port) => vec![
-                SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port),
-                SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), port),
-            ]
-            .into_iter(),
+            Address::Localhost(ipv6_enabled, port) => {
+                if ipv6_enabled {
+                    vec![
+                        SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port),
+                        SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), port),
+                    ]
+                    .into_iter()
+                } else {
+                    vec![SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port)].into_iter()
+                }
+            }
             Address::SocketAddr(s) => vec![s].into_iter(),
         }
     }
 }
 
-impl FromStr for Address {
-    type Err = anyhow::Error;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+impl Address {
+    fn from_str(ipv6_enabled: bool, s: &str) -> anyhow::Result<Self> {
```

Comment:
**bikeshed:** I wonder if we should use different name entirely since we're not implementing the FromStr trait anymore and it's not really making an address from a string... Maybe just `new`

Reply:
:+1: